src/components/Molecules/Task/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Checkbox } from "../../Atoms/Checkbox/index";
+import { EditButton } from "../../Atoms/EditButton/index";
+import { Input } from "../../Atoms/Input/index";
+import styled from "styled-components";
+import COLOR from "../../../variables/color";
+import TEXT from "../../../variables/texts";
+import FONT_FAMILY from "../../../variables/font_family";
+
+export const Task = ({
+  taskName = "",
+  onTaskComplete,
+  defaultIsEditing = false,
+  onTaskNameChange,
+}) => {
+  const [isEditing, setIsEditing] = useState(defaultIsEditing);
+
+  const onEditComplete = (value) => {
+    setIsEditing(false);
+    onTaskNameChange(value);
+  };
+
+  const onEditButtonClick = () => {
+    setIsEditing(true);
+  };
+
+  return (
+    <StyledContainer>
+      <StyledCheckboxContainer>
+        <Checkbox onClick={onTaskComplete} />
+      </StyledCheckboxContainer>
+      {isEditing ? (
+        <Input onEditComplete={onEditComplete} defaultValue={taskName} />
+      ) : (
+        <StyledNameAndButtonContainer>
+          <StyledTaskName>{taskName}</StyledTaskName>
+          <StyledEditButtonContainer>
+            <EditButton onClick={onEditButtonClick} />
+          </StyledEditButtonContainer>
+        </StyledNameAndButtonContainer>
+      )}
+    </StyledContainer>
+  );
+};
+export default Task;
+
+const StyledContainer = styled.div`
+  padding: 2px 6px;
+  display: flex;
+  align-items: flex-start;
+  gap: 10px;
+  flex: 1 0 0;
+`;
+
+const StyledCheckboxContainer = styled.div`
+  display: flex;
+  width: 20px;
+  height: 20px;
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledNameAndButtonContainer = styled.div`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  gap: 10px;
+  overflow: hidden;
+`;
+
+const StyledTaskName = styled.div`
+  color: ${COLOR.LIGHT_GRAY};
+  font-family: ${FONT_FAMILY.NOTO_SANS};
+  ${TEXT.S}
+  margin-right: 10px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```

Comment:
```suggestion
  text-overflow: ellipsis;
  flex: 1;
```

非編集時にテキスト部分が幅いっぱいに広がるように修正をお願いします :pray:

<img width="330" alt="image" src="https://github.com/Akidesign/aki-fukushima-frontend-basic-2022-todoapp/assets/83479175/5ed3e075-9b5b-4070-b1dd-f7df87dadc79">

<img width="330" alt="image" src="https://github.com/Akidesign/aki-fukushima-frontend-basic-2022-todoapp/assets/83479175/2848ae1f-afd8-47fe-a308-7544fc30a046">